tests/subsys/llext/simple/src/test_llext_simple.c

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2023 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/ztest.h>
+#include <zephyr/kernel.h>
+#include <zephyr/llext/llext.h>
+#include <zephyr/llext/buf_loader.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(test_llext_simple);
+
+
+#ifdef CONFIG_LLEXT_STORAGE_WRITABLE
+#define LLEXT_CONST const
+#else
+#define LLEXT_CONST
+#endif
```

Comment:
I'm hesitant to move it in this PR as its not showing any use cases outside this test, but I'd happily accept a PR showing its use in say the shell sample and moving it to llext.h

Reply:
Understood. Still, would love this even more if it was not flipped over... :hand_over_mouth: 
Causes a failure on `qemu_cortex_m3:llext.simple.writable` because it's trying to link by modifing code in Flash.